src/Support/Collection/KeyValueCollection.php

```diff
@@ -97,4 +97,11 @@ public function remove(string $key): KeyValueCollectionInterface
 
         return $this;
     }
+
+    public function pull(string $key, mixed $defaultValue = null): mixed
```

Comment:
The interface change is missing (`src/Support/Collection/KeyValueCollectionInterface.php`), the documentation (`docs/03-collections.md`) and a test.